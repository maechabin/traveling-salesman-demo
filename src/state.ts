import { State } from './state.model';

const state: State = {
  departure: {
    title: '山王パークタワー',
    lat: 35.6731652,
    lng: 139.7407961,
    label: '発',
  },
  arival: {
    title: '六本木ヒルズ森タワー',
    lat: 35.6604638,
    lng: 139.72924869999997,
    label: '着',
  },
  routes: [
    {
      id: 1,
      title: 'Wコンフォートタワーズ EAST',
      lat: 35.6471312,
      lng: 139.80113460000007,
      label: 'A',
      sortId: 0,
    },
    {
      id: 2,
      title: 'セルリアンタワー',
      lat: 35.656325,
      lng: 139.699608,
      label: 'B',
      sortId: 0,
    },
    {
      id: 3,
      title: '新宿センタービル',
      lat: 35.6917502,
      lng: 139.6955153,
      label: 'C',
      sortId: 0,
    },
    {
      id: 4,
      title: 'JPタワー',
      lat: 35.679799,
      lng: 139.764545,
      label: 'D',
      sortId: 0,
    },
    {
      id: 5,
      title: '新宿パークタワー',
      lat: 35.68564,
      lng: 139.69094999999993,
      label: 'E',
      sortId: 0,
    },
    {
      id: 6,
      title: '住友不動産新宿グランドタワー',
      lat: 35.6957954,
      lng: 139.69038920000003,
      label: 'F',
      sortId: 0,
    },
    {
      id: 7,
      title: '勝どきビュータワー',
      lat: 35.6595554,
      lng: 139.77651500000002,
      label: 'G',
      sortId: 0,
    },
    {
      id: 8,
      title: 'サンシャイン６０',
      lat: 35.7291213,
      lng: 139.7191322,
      label: 'H',
      sortId: 0,
    },
  ],
  initialFlag: true,
  choosingRouteStartFlag: false,
  choosingRouteFinishFlag: false,
  viewAnswerFlag: false,
  currentSortId: 8, // routes（経路）の要素数を指定する
  transport: 'car',
  expressway: 'no',
  traffic: 'standard',
  departureTime: new Date(Date.now() + 1000 * 60 * 60 * 24 * 3),
  gross: {
    distance: 0, // 総距離
    duration: 0, // 総時間
  },
  answerGross: {
    distance: 0, // 正解の総距離
    duration: 0, // 正解の総時間
  },
  answerWaypointOrder: [], // 正解の経路の表示順
};

export default state;
