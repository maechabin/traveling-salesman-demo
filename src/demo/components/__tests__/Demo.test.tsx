import React from 'react';
import { shallow } from 'enzyme';

import Demo from '../Demo';
import { Step } from '../../../state.model';

describe('<Demo />', () => {
  it('questionStepがStep.Answerの場合、<DemoAnswer />がレンダリングされること', () => {
    // setup
    const props = {
      questionStep: Step.Answer,
    } as any;

    // exercise
    const wrapper = shallow(<Demo {...props} />);

    // verify
    expect(wrapper.find('DemoAnswer').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('questionStepがStep.Initialの場合、<DemoDescription />がレンダリングされること', () => {
    // setup
    const props = {
      questionStep: Step.Initial,
    } as any;

    // exercise
    const wrapper = shallow(<Demo {...props} />);

    // verify
    expect(wrapper.find('DemoDescription').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('questionStepがStep.Initialの場合、<DemoEdit />がレンダリングされること', () => {
    // setup
    const props = {
      questionStep: Step.Edit,
    } as any;

    // exercise
    const wrapper = shallow(<Demo {...props} />);

    // verify
    expect(wrapper.find('DemoEdit').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
