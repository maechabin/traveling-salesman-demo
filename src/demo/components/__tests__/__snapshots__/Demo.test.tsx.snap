// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Demo /> questionStepがStep.Answerの場合、<DemoAnswer />がレンダリングされること 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Demo
    questionStep={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <main
        className="DemoMain"
      >
        <DemoQuestion
          questionStep={5}
        />
        <DemoAnswer
          questionStep={5}
        />
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DemoQuestion
            questionStep={5}
          />,
          <DemoAnswer
            questionStep={5}
          />,
        ],
        "className": "DemoMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionStep": 5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionStep": 5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <main
          className="DemoMain"
        >
          <DemoQuestion
            questionStep={5}
          />
          <DemoAnswer
            questionStep={5}
          />
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DemoQuestion
              questionStep={5}
            />,
            <DemoAnswer
              questionStep={5}
            />,
          ],
          "className": "DemoMain",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "questionStep": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "questionStep": 5,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Demo /> questionStepがStep.Initialの場合、<DemoDescription />がレンダリングされること 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Demo
    questionStep={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <main
        className="DemoMain"
      >
        <DemoQuestion
          questionStep={0}
        />
        <DemoDescription
          questionStep={0}
        />
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DemoQuestion
            questionStep={0}
          />,
          <DemoDescription
            questionStep={0}
          />,
        ],
        "className": "DemoMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionStep": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionStep": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <main
          className="DemoMain"
        >
          <DemoQuestion
            questionStep={0}
          />
          <DemoDescription
            questionStep={0}
          />
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DemoQuestion
              questionStep={0}
            />,
            <DemoDescription
              questionStep={0}
            />,
          ],
          "className": "DemoMain",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "questionStep": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "questionStep": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Demo /> questionStepがStep.Initialの場合、<DemoEdit />がレンダリングされること 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Demo
    questionStep={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <main
        className="DemoMain"
      >
        <DemoQuestion
          questionStep={2}
        />
        <DemoEdit
          questionStep={2}
        />
      </main>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DemoQuestion
            questionStep={2}
          />,
          <DemoEdit
            questionStep={2}
          />,
        ],
        "className": "DemoMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionStep": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "questionStep": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <main
          className="DemoMain"
        >
          <DemoQuestion
            questionStep={2}
          />
          <DemoEdit
            questionStep={2}
          />
        </main>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DemoQuestion
              questionStep={2}
            />,
            <DemoEdit
              questionStep={2}
            />,
          ],
          "className": "DemoMain",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "questionStep": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "questionStep": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
